### PR TITLE
Add tests for sidecar containers

### PR DESCRIPTION
**Pull request by idlewis**
Sidecar container tests as per issue #305

### CODE REVIEW

File: bundle/tests/scorecard/kuttl/sidecar-deployment/02-assert.yaml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+apiVersion: v1
+kind: Pod
+spec:
+  containers:
+  - name: app
```
Comment (leochr) [inline bundle/tests/scorecard/kuttl/sidecar-deployment/02-assert.yaml:5]: It'll be good to add the label `app.kubernetes.io/instance` to ensure that it's the right Pod and not a Pod leftbehind from previous test scenarios (i.e. init-container)

File: bundle/tests/scorecard/kuttl/sidecar-stateful/02-assert.yaml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+apiVersion: v1
+kind: Pod
+spec:
+  containers:
+  - name: app
```
Comment (leochr) [inline bundle/tests/scorecard/kuttl/sidecar-stateful/02-assert.yaml:5]: Same comment about Pod label.

**Review by leochr — COMMENTED** [review]
@idlewis Looks great! Thanks. Added a minor comment about using label for certain Pods to ensure it was the right Pod.

**Comment by idlewis** [comment]
@leochr updated as requested

**Review by leochr — APPROVED** [review]
Thanks

**Pull request merged**